paasta_tools/cli/cmds/spark_run.py

```diff
@@ -326,6 +353,54 @@ def get_spark_conf_str(
     volumes,
 ):
     spark_conf = list()
+
+    base_args = {
+        'spark.cores.max': '4',
+        'spark.executor.cores': '2',
+        'spark.executor.memory': '4g',
+    }
+
+    if args.spark_args:
+        spark_args = args.spark_args.split()
+        for spark_arg in spark_args:
+            fields = spark_arg.split('=')
+            if len(fields) != 2:
+                paasta_print(
+                    PaastaColors.red(
+                        "Spark option %s is not in format option=value." % spark_arg,
+                    ),
+                    file=sys.stderr,
+                )
+                sys.exit(1)
+            base_args[fields[0]] = fields[1]
+
+    if int(base_args['spark.cores.max']) < int(base_args['spark.executor.cores']):
+        paasta_print(
+            PaastaColors.red(
+                "Total number of cores %s is less than per-executor cores %s." % (
+                    base_args['spark.cores.max'],
+                    base_args['spark.executor.cores'],
+                ),
+            ),
+            file=sys.stderr,
+        )
+        sys.exit(1)
+
+    exec_mem = base_args['spark.executor.memory']
+    if exec_mem[-1] != 'g' or not exec_mem[:-1].isdigit() or int(exec_mem[:-1]) > 32:
+        paasta_print(
+            PaastaColors.red(
+                "Executor memory %s not in format dg (d<=32)." % (
+                    base_args['spark.executor.memory'],
+                ),
+            ),
+            file=sys.stderr,
+        )
+        sys.exit(1)
+
+    for opt, val in base_args.items():
+        spark_conf.append('--conf %s=%s' % (opt, val))
+
     spark_conf.append('--conf spark.app.name=%s' % container_name)
```

Comment:
why not include stuff like this in `base_args`?  Also what happens if the user tries to specify `spark.app.name` themselves?  Will the fact that we're appending this afterwards override their value?